src/api/app/models/workflow/step/submit_request.rb

```diff
@@ -0,0 +1,118 @@
+                                            type: 'submit')
+    bs_request = BsRequest.new(bs_request_actions: [bs_request_action],
+                               description: bs_request_description)
+    Pundit.authorize(@token.executor, bs_request, :create?)
+
+    begin
+      bs_request.save!
+    rescue MaintenanceHelper::MissingAction
+      raise 'Unable to submit, sources are unchanged'
+    rescue Project::Errors::UnknownObjectError
+      raise "Unable to submit: The source of package #{source_project_name}/#{source_package_name} is broken"
+    rescue APIError, ActiveRecord::RecordInvalid => e
+      raise e.message
+    rescue Backend::Error => e
+      raise e.summary
+    end
+
+    create_or_update_subscriptions(bs_request: bs_request)
+    (@request_numbers_and_state_for_artifacts["#{bs_request.state}"] ||= []) << bs_request.number
+    bs_request
+  end
+
+  def supersede_previous_and_submit_request
+    # Fetch current open submit request which are going to be superseded
+    # after the new sumbit request is created
+    requests_to_be_superseded = submit_requests_with_same_target_and_source
+    new_submit_request = submit_package
+
+    return if requests_to_be_superseded.blank?
+
+    requests_to_be_superseded.each do |submit_request|
+      # Authorization happens on model level
+      request = BsRequest.find_by_number!(submit_request.number)
+      request.change_state(newstate: 'superseded',
+                           reason: "Superseded by request #{new_submit_request.number}",
+                           superseded_by: new_submit_request.number)
+      (@request_numbers_and_state_for_artifacts["#{request.state}"] ||= []) << request.number
+    end
+  end
+
+  def revoke_submit_requests
+    return if submit_requests_with_same_target_and_source.blank?
+
+    submit_requests_with_same_target_and_source.each do |submit_request|
+      next unless Pundit.authorize(@token.executor, submit_request, :revoke_request?)
+
+      submit_request.change_state(newstate: 'revoked', comment: "Revoke as #{workflow_run.event_source_url} got closed")
+      (@request_numbers_and_state_for_artifacts["#{submit_request.state}"] ||= []) << submit_request.number
+    end
+  end
+
+  def submit_requests_with_same_target_and_source
+    BsRequest.list({ project: step_instructions[:target_project],
+                     source_project: step_instructions[:source_project],
+                     package: step_instructions[:source_package],
+                     types: 'submit', states: ['new', 'review', 'declined'] })
+  end
+
+  def source_package
+    Package.get_by_project_and_name(source_project_name, source_package_name, follow_multibuild: true)
+  rescue Project::Errors::UnknownObjectError, Package::Errors::UnknownObjectError
+    # We rely on Package.get_by_project_and_name since it's the only way to work with multibuild packages.
+    raise "The source project or package '#{source_project_name}/#{source_package_name}' does not exist"
```

Comment:
Again raising an `RuntimeError` that no one will catch.

Reply:
Removed